src/components/ProgressBar/index.tsx

```diff
@@ -1,8 +1,54 @@
-import {
-  ProgressBarContainer,
-  ProgressBarBody,
-  ProgressCircle,
-} from './styles';
+import { useCallback, useContext } from 'react';
+import { H4 } from '@/styles/text';
+import { FlowData } from '@/types/misc';
+import { useOnboardingNavigation } from '@/utils/hooks';
+import { OnboardingContext } from '@/utils/OnboardingProvider';
+import * as Styles from './styles';
+
+// helper component
+function ProgressCircle({
+  step,
+  idx,
+  progress,
+}: {
+  step: FlowData;
+  idx: number;
+  progress: number;
+}) {
+  const onboarding = useContext(OnboardingContext);
+  if (!onboarding)
+    throw new Error(
+      'Fatal: onboarding progress should be placed inside onboarding context provider',
+    );
+
+  const clickable = idx !== progress && onboarding.progress >= idx;
+  const disabled = onboarding.progress <= idx && idx !== progress;
+
+  const { flow: onboardingFlow } = onboarding;
+  const { ebbTo } = useOnboardingNavigation();
+
+  const onClick = useCallback(() => {
+    ebbTo(`/onboarding/${onboardingFlow[idx].url}`);
+  }, [ebbTo, onboardingFlow, idx]);
+
+  return (
+    <Styles.ProgressCircleContainer
```

Comment:
why are these different from all other files with styled components? like why is this styles. and everything else doesn't have this

Reply:
these are local styles